### ISSUE TITLE
Apply PSR-2 coding style

### ISSUE THREAD

**Issue by thomaszbz** [mblaschke/TYPO3-metaseo#35]
PSR-2 is claimed for this project, but still not applied completely for existing code. I suggest to apply it before releasing the 2.0 version.

The reason for that is that for the moment there are not many forks of this project out there. Literally speaking, nearly all branches are clean. As soon as this changes, more and more code is changed in other branches leading to conflicts when merging them again after applying coding style. In the end, doing it now could avoid a lot of pain in the long run (aka the typo3 dilemma).

Plus, there's much lesser code to change in later 2.0 production versions leading to easier tracking of fixes. Currently, it's not easy for the reader to decide what is done for coding style and what is the actual fix/change.

**Comment by thomaszbz** [comment]
current status as of bd807f1ad7fc26c7eef284abb550ec7fed9f4fa3 : 

    /usr/bin/phpcs --standard=PSR2 --report=summary --report-width=120 .
    
    PHP CODE SNIFFER REPORT SUMMARY
    -----------------------------------------------------------------------------------------------
    FILE                                                                           ERRORS  WARNINGS
    -----------------------------------------------------------------------------------------------
    metaseo/Classes/Backend/Ajax/PageAjax.php                                      63      3
    metaseo/Classes/Backend/Ajax/SitemapAjax.php                                   16      1
    metaseo/Classes/Backend/Ajax/AbstractAjax.php                                  21      1
    metaseo/Classes/Backend/Module/AbstractStandardModule.php                      2       0
    metaseo/Classes/Backend/Module/AbstractTreeModule.php                          2       0
    metaseo/Classes/Backend/Module/AbstractModule.php                              13      0
    metaseo/Classes/Backend/Validator/ValidatorImport.php                          3       0
    metaseo/Classes/Backend/Validator/FloatValidator.php                           5       0
    metaseo/Classes/Command/MetaseoCommandController.php                           15      0
    metaseo/Classes/Controller/BackendControlCenterController.php                  10      3
    metaseo/Classes/Controller/BackendPageSeoController.php                        17      8
    metaseo/Classes/Controller/BackendRootSettingsController.php                   10      3
    metaseo/Classes/Controller/BackendSitemapController.php                        25      6
    metaseo/Classes/Hook/ClearCacheHook.php                                        4       0
    metaseo/Classes/Hook/Extension/NewsExtension.php                               3       0
    metaseo/Classes/Hook/Extension/TtnewsExtension.php                             4       0
    metaseo/Classes/Hook/HttpHook.php                                              4       0
    metaseo/Classes/Hook/SitemapIndexHook.php                                      49      0
    metaseo/Classes/Hook/TCA/RobotsTxtDefault.php                                  4       5
    metaseo/Classes/Page/Part/AbstractPart.php                                     4       0
    metaseo/Classes/Page/Part/FooterPart.php                                       29      2
    metaseo/Classes/Page/Part/PagetitlePart.php                                    19      1
    metaseo/Classes/Page/Part/MetatagPart.php                                      54      31
    metaseo/Classes/Page/RobotsTxtPage.php                                         10      2
    metaseo/Classes/Page/AbstractPage.php                                          9       1
    metaseo/Classes/Page/SitemapTxtPage.php                                        7       0
    metaseo/Classes/Page/SitemapXmlPage.php                                        7       0
    metaseo/Classes/Scheduler/Task/GarbageCollectionTask.php                       4       0
    metaseo/Classes/Scheduler/Task/AbstractSitemapTask.php                         13      0
    metaseo/Classes/Scheduler/Task/AbstractTask.php                                30      0
    metaseo/Classes/Scheduler/Task/SitemapTxtTask.php                              6       0
    metaseo/Classes/Scheduler/Task/SitemapXmlTask.php                              8       0
    metaseo/Classes/Sitemap/Generator/AbstractGenerator.php                        10      0
    metaseo/Classes/Sitemap/Generator/TxtGenerator.php                             4       0
    metaseo/Classes/Sitemap/Generator/XmlGenerator.php                             8       3
    metaseo/Classes/Utility/BackendUtility.php                                     8       0
    metaseo/Classes/Utility/CacheUtility.php                                       29      0
    metaseo/Classes/Utility/ConsoleUtility.php                                     118     0
    metaseo/Classes/Utility/DatabaseUtility.php                                    73      1
    metaseo/Classes/Utility/FrontendUtility.php                                    59      1
    metaseo/Classes/Utility/GeneralUtility.php                                     54      1
    metaseo/Classes/Utility/RootPageUtility.php                                    7       0
    metaseo/Classes/Utility/SitemapUtility.php                                     20      1
    metaseo/Classes/Connector.php                                                  19      0
    metaseo/Configuration/TCA/Overrides/pages.php                                  218     3
    metaseo/Configuration/TCA/Overrides/pages_language_overlay.php                 91      3
    metaseo/Configuration/TCA/tx_metaseo_setting_root.php                          130     9
    metaseo/Resources/Public/Backend/Css/Default.css                               20      0
    metaseo/Resources/Public/Backend/JavaScript/MetaSeo.metaeditor.js              12      0
    metaseo/Resources/Public/Backend/JavaScript/MetaSeo.overview.js                30      0
    metaseo/Resources/Public/Backend/JavaScript/MetaSeo.sitemap.js                 6       0
    metaseo/Resources/Public/Frontend/JavaScript/GoogleAnalyticsTrackDownload.js   42      0
    metaseo/class.ext_update.php                                                   26      0
    metaseo/ext_emconf.php                                                         36      2
    metaseo/ext_localconf.php                                                      0       18
    metaseo/ext_tables.php                                                         4       0
    -----------------------------------------------------------------------------------------------
    A TOTAL OF 1494 ERROR(S) AND 109 WARNING(S) WERE FOUND IN 56 FILE(S)
    -----------------------------------------------------------------------------------------------

**Comment by thomaszbz** [comment]
Applying

    php-cs-fixer fix /home/thomas/75_phpstorm/metaseo/ --level=psr2

looks promising:

    /usr/bin/phpcs --standard=PSR2 --report=summary --report-width=120 .

    PHP CODE SNIFFER REPORT SUMMARY
    -----------------------------------------------------------------------------------------------
    FILE                                                                           ERRORS  WARNINGS
    -----------------------------------------------------------------------------------------------

    metaseo/Classes/Backend/Ajax/PageAjax.php                                      8       3
    metaseo/Classes/Backend/Ajax/SitemapAjax.php                                   0       1
    metaseo/Classes/Backend/Ajax/AbstractAjax.php                                  1       1
    metaseo/Classes/Backend/Validator/ValidatorImport.php                          2       0
    metaseo/Classes/Controller/BackendControlCenterController.php                  1       3
    metaseo/Classes/Controller/BackendPageSeoController.php                        3       8
    metaseo/Classes/Controller/BackendRootSettingsController.php                   1       3
    metaseo/Classes/Controller/BackendSitemapController.php                        3       6
    metaseo/Classes/Hook/SitemapIndexHook.php                                      2       0
    metaseo/Classes/Hook/TCA/RobotsTxtDefault.php                                  0       5
    metaseo/Classes/Page/Part/FooterPart.php                                       7       2
    metaseo/Classes/Page/Part/PagetitlePart.php                                    1       1
    metaseo/Classes/Page/Part/MetatagPart.php                                      0       31
    metaseo/Classes/Page/RobotsTxtPage.php                                         0       2
    metaseo/Classes/Page/AbstractPage.php                                          0       1
    metaseo/Classes/Scheduler/Task/AbstractSitemapTask.php                         1       0
    metaseo/Classes/Scheduler/Task/SitemapTxtTask.php                              1       0
    metaseo/Classes/Scheduler/Task/SitemapXmlTask.php                              2       0
    metaseo/Classes/Sitemap/Generator/XmlGenerator.php                             0       3
    metaseo/Classes/Utility/DatabaseUtility.php                                    0       1
    metaseo/Classes/Utility/FrontendUtility.php                                    0       1
    metaseo/Classes/Utility/GeneralUtility.php                                     0       1
    metaseo/Classes/Utility/SitemapUtility.php                                     0       1
    metaseo/Configuration/TCA/Overrides/pages.php                                  0       11
    metaseo/Configuration/TCA/Overrides/pages_language_overlay.php                 0       3
    metaseo/Configuration/TCA/tx_metaseo_setting_root.php                          0       14
    metaseo/Resources/Public/Backend/Css/Default.css                               20      0
    metaseo/Resources/Public/Backend/JavaScript/MetaSeo.metaeditor.js              12      0
    metaseo/Resources/Public/Backend/JavaScript/MetaSeo.overview.js                30      0
    metaseo/Resources/Public/Backend/JavaScript/MetaSeo.sitemap.js                 6       0
    metaseo/Resources/Public/Frontend/JavaScript/GoogleAnalyticsTrackDownload.js   42      0
    metaseo/class.ext_update.php                                                   3       0
    metaseo/ext_emconf.php                                                         0       2
    metaseo/ext_localconf.php                                                      0       18
    -----------------------------------------------------------------------------------------------
    A TOTAL OF 146 ERROR(S) AND 122 WARNING(S) WERE FOUND IN 34 FILE(S)
    -----------------------------------------------------------------------------------------------

**Comment by thomaszbz** [comment]
class tx_metaseo_backend_validation_float is deprecated since at least 2014-08-31. I also found no usage of the class. Can I drop the class?

**Comment by thomaszbz** [comment]
There's been a lot of lines with too many characters. Therefore, I tried  removing fully qualified names in c41a342 :

    /usr/bin/phpcs --standard=PSR2 --report=summary --report-width=120 .

    PHP CODE SNIFFER REPORT SUMMARY
    -----------------------------------------------------------------------------------------------
    FILE                                                                           ERRORS  WARNINGS
    -----------------------------------------------------------------------------------------------

    metaseo/Classes/Backend/Ajax/AbstractAjax.php                                  1       0
    metaseo/Classes/Backend/Ajax/PageAjax.php                                      8       2
    metaseo/Classes/Backend/Validator/ValidatorImport.php                          2       0
    metaseo/Classes/Controller/BackendRootSettingsController.php                   1       1
    metaseo/Classes/Controller/BackendPageSeoController.php                        3       7
    metaseo/Classes/Controller/BackendSitemapController.php                        3       3
    metaseo/Classes/Controller/BackendControlCenterController.php                  1       1
    metaseo/Classes/Hook/TCA/RobotsTxtDefault.php                                  0       4
    metaseo/Classes/Hook/SitemapIndexHook.php                                      2       0
    metaseo/Classes/Page/Part/FooterPart.php                                       7       2
    metaseo/Classes/Page/Part/PagetitlePart.php                                    1       1
    metaseo/Classes/Page/Part/MetatagPart.php                                      0       31
    metaseo/Classes/Page/RobotsTxtPage.php                                         0       1
    metaseo/Classes/Scheduler/Task/SitemapTxtTask.php                              1       0
    metaseo/Classes/Scheduler/Task/SitemapXmlTask.php                              2       0
    metaseo/Classes/Scheduler/Task/AbstractSitemapTask.php                         1       0
    metaseo/Classes/Sitemap/Generator/XmlGenerator.php                             0       2
    metaseo/Classes/Utility/DatabaseUtility.php                                    0       1
    metaseo/Classes/Utility/SitemapUtility.php                                     0       1
    metaseo/Configuration/TCA/Overrides/pages_language_overlay.php                 0       3
    metaseo/Configuration/TCA/Overrides/pages.php                                  0       11
    metaseo/Configuration/TCA/tx_metaseo_setting_root.php                          0       14
    metaseo/Resources/Public/Backend/Css/Default.css                               20      0
    metaseo/Resources/Public/Backend/JavaScript/MetaSeo.metaeditor.js              12      0
    metaseo/Resources/Public/Backend/JavaScript/MetaSeo.overview.js                30      0
    metaseo/Resources/Public/Backend/JavaScript/MetaSeo.sitemap.js                 6       0
    metaseo/Resources/Public/Frontend/JavaScript/GoogleAnalyticsTrackDownload.js   42      0
    metaseo/ext_localconf.php                                                      0       18
    metaseo/ext_emconf.php                                                         0       2
    metaseo/class.ext_update.php                                                   3       0
    -----------------------------------------------------------------------------------------------
    A TOTAL OF 146 ERROR(S) AND 105 WARNING(S) WERE FOUND IN 30 FILE(S)
    -----------------------------------------------------------------------------------------------

ok, this was good for nothing in terms of coding style, thus still making the code much cleaner.

**Comment by thomaszbz** [comment]
Now, that was a good one in 1bd547d , fixing warnings mainly

    /usr/bin/phpcs --standard=PSR2 --report=summary --report-width=120 .

    PHP CODE SNIFFER REPORT SUMMARY
    -----------------------------------------------------------------------------------------------
    FILE                                                                           ERRORS  WARNINGS
    -----------------------------------------------------------------------------------------------
    metaseo/Classes/Backend/Ajax/AbstractAjax.php                                  1       0
    metaseo/Classes/Backend/Ajax/PageAjax.php                                      8       1
    metaseo/Classes/Backend/Validator/ValidatorImport.php                          2       0
    metaseo/Classes/Controller/BackendControlCenterController.php                  1       0
    metaseo/Classes/Controller/BackendPageSeoController.php                        3       0
    metaseo/Classes/Controller/BackendSitemapController.php                        9       0
    metaseo/Classes/Controller/BackendRootSettingsController.php                   1       0
    metaseo/Classes/Hook/SitemapIndexHook.php                                      2       0
    metaseo/Classes/Page/Part/PagetitlePart.php                                    1       0
    metaseo/Classes/Page/Part/FooterPart.php                                       2       0
    metaseo/Classes/Scheduler/Task/SitemapTxtTask.php                              1       0
    metaseo/Classes/Scheduler/Task/SitemapXmlTask.php                              2       0
    metaseo/Classes/Scheduler/Task/AbstractSitemapTask.php                         1       0
    metaseo/Classes/Utility/DatabaseUtility.php                                    0       1
    metaseo/Configuration/TCA/Overrides/pages_language_overlay.php                 0       3
    metaseo/Configuration/TCA/Overrides/pages.php                                  0       11
    metaseo/Configuration/TCA/tx_metaseo_setting_root.php                          0       14
    metaseo/Resources/Public/Backend/Css/Default.css                               20      0
    metaseo/Resources/Public/Backend/JavaScript/MetaSeo.metaeditor.js              12      0
    metaseo/Resources/Public/Backend/JavaScript/MetaSeo.overview.js                30      0
    metaseo/Resources/Public/Backend/JavaScript/MetaSeo.sitemap.js                 6       0
    metaseo/Resources/Public/Frontend/JavaScript/GoogleAnalyticsTrackDownload.js   42      0
    metaseo/ext_localconf.php                                                      0       18
    metaseo/ext_emconf.php                                                         0       2
    metaseo/class.ext_update.php                                                   3       0
    -----------------------------------------------------------------------------------------------
    A TOTAL OF 147 ERROR(S) AND 50 WARNING(S) WERE FOUND IN 25 FILE(S)
    -----------------------------------------------------------------------------------------------

**Comment by thomaszbz** [comment]
Now, that was a good one in  6b57c1e

    /usr/bin/phpcs --standard=PSR2 --report=summary --report-width=120 .

    PHP CODE SNIFFER REPORT SUMMARY
    -----------------------------------------------------------------------------------------------
    FILE                                                                           ERRORS  WARNINGS
    -----------------------------------------------------------------------------------------------
    metaseo/Classes/Backend/Validator/ValidatorImport.php                          2       0
    metaseo/Classes/Hook/SitemapIndexHook.php                                      2       0
    metaseo/Classes/Page/Part/FooterPart.php                                       1       0
    metaseo/Classes/Scheduler/Task/SitemapTxtTask.php                              1       0
    metaseo/Classes/Scheduler/Task/SitemapXmlTask.php                              2       0
    metaseo/Classes/Scheduler/Task/AbstractSitemapTask.php                         1       0
    metaseo/Resources/Public/Backend/Css/Default.css                               20      0
    metaseo/Resources/Public/Backend/JavaScript/MetaSeo.metaeditor.js              12      0
    metaseo/Resources/Public/Backend/JavaScript/MetaSeo.overview.js                30      0
    metaseo/Resources/Public/Backend/JavaScript/MetaSeo.sitemap.js                 6       0
    metaseo/Resources/Public/Frontend/JavaScript/GoogleAnalyticsTrackDownload.js   42      0
    metaseo/class.ext_update.php                                                   2       0
    metaseo/ext_emconf.php                                                         0       1
    -----------------------------------------------------------------------------------------------
    A TOTAL OF 121 ERROR(S) AND 1 WARNING(S) WERE FOUND IN 13 FILE(S)
    -----------------------------------------------------------------------------------------------

All remaining issues in .php files cannot be fixed as they come from typo3 requirements.

**Comment by thomaszbz** [comment]
Also fixed most of the javascript issues in 2fa725e .

In terms of coding style the rest is required by typo3. There's still some files containing tabs in directories Documentation and Resources only.

    /usr/bin/phpcs --standard=PSR2 --report=summary --report-width=120 .

    PHP CODE SNIFFER REPORT SUMMARY
    -----------------------------------------------------------------------------------------------
    FILE                                                                           ERRORS  WARNINGS
    -----------------------------------------------------------------------------------------------
    metaseo/Classes/Backend/Validator/ValidatorImport.php                          2       0
    metaseo/Classes/Hook/SitemapIndexHook.php                                      2       0
    metaseo/Classes/Page/Part/FooterPart.php                                       1       0
    metaseo/Classes/Scheduler/Task/SitemapTxtTask.php                              1       0
    metaseo/Classes/Scheduler/Task/SitemapXmlTask.php                              2       0
    metaseo/Classes/Scheduler/Task/AbstractSitemapTask.php                         1       0
    metaseo/class.ext_update.php                                                   2       0
    metaseo/ext_emconf.php                                                         0       1
    -----------------------------------------------------------------------------------------------
    A TOTAL OF 11 ERROR(S) AND 1 WARNING(S) WERE FOUND IN 8 FILE(S)
    -----------------------------------------------------------------------------------------------

**Comment by thomaszbz** [comment]
In general I hope that all the changes for this ticket contain no change in behaviour. Most of it is syntax-only that has changed.

Please review the changes in javascript files ( 2fa725e ) carefully before merging. I did not test them at all.
